src/Altinn.Notifications.Email.Core/Configuration/ServiceCollectionExtensions.cs

```diff
@@ -16,7 +16,15 @@ public static class ServiceCollectionExtensions
     /// <returns>The given service collection.</returns>
     public static IServiceCollection AddCoreServices(this IServiceCollection services, IConfiguration config)
     {
-        services.AddSingleton<IEmailService, EmailService>();
+        TopicSettings topicSettings = config!.GetSection("KafkaSettings").Get<TopicSettings>()!;
```

Comment:
I don't want us to do this in core. Core should not know about Kafka.

Reply:
Think it's hard to get around this. We could of course have some sort of service masking that it is kafka, but core handles the business logic and is the one to determine which topic something should be postet to based on success or failure of some sort of action that core has orchestrated.

Reply:
I'm not a fan of binding KafkaSettings to TopicSettings. It's a smell related to the original comment. 

I think your idea of introducing a new service is the right choice.